src/xma/src/xmaapi/xmaapi.cpp

```diff
@@ -267,6 +271,7 @@ void xma_thread2(uint32_t hw_dev_index) {
             xrt_device_obj.get_handle()->exec_wait(100);
         }
 
+        std::lock_guard<std::mutex> lock(g_xma_singleton->m_mutex);
```

Comment:
```suggestion
        std::lock_guard lock(g_xma_singleton->m_mutex);
```
In C++17 we trust.

Reply:
Why though? The rest of the code uses the lock guard like this, want to stay consistent.